github/event_types.go

```diff
@@ -268,6 +270,19 @@ type LabelEvent struct {
 	Installation *Installation `json:"installation,omitempty"`
 }
 
+// MarketplacePurchaseEvent is triggered when a user purchases, cancels, or changes
+// their GitHub Marketplace plan.
+// Webhook event name "marketplace_purchase".
+//
+// Github API docs: https://developer.github.com/v3/activity/events/types/#marketplacepurchaseevent
+type MarketplacePurchaseEvent struct {
+	Action                      *string              `json:"action"`
+	EffectiveDate               *time.Time           `json:"effective_date"`
+	MarketplacePurchase         *MarketplacePurchase `json:"marketplace_purchase"`
+	PreviousMarketplacePurchase *MarketplacePurchase `json:"previous_marketplace_purchase"`
+	Sender                      *User                `json:"sender,omitempty"`
+}
```

Comment:
You'll need to break the fields into two sections. The event API payload fields (of which, there's only one, `action`):

![image](https://user-images.githubusercontent.com/1924134/32993806-5c53e4e8-cd2b-11e7-89e6-d178f4d2419f.png)

And Webhook payload fields, which are all the rest. You should put them under a `// The following fields are only populated by Webhook events.` comment. Something like this:

```Go
type MarketplacePurchaseEvent struct {
	// Action is the action that was performed. Possible values are:
	// "purchased", "cancelled", or "changed".
	Action *string `json:"action,omitempty"`

	// The following fields are only populated by Webhook events.
	...
}
```

See structs right above and below the one you added for reference.

Reply:
Additionally, you'll need to add an implicit `Installation *Installation `json:"installation,omitempty"` field. Its existence is documented at https://developer.github.com/webhooks/#payloads. Put it at the very bottom; see other events in this file as reference.

Reply:
Okay @shurcooL, will fix these. thanks :+1:  .